### PR TITLE
Add docs for i18n and update i18n.js lib

### PR DESCRIPTION
**Pull request by alicoding**
Add documentation on how to localize a component and also update i18n.js
library for more methods

### CODE REVIEW

**Comment by alicoding** [comment]
@k88hudson R?

File: src/lib/i18n.js [hunk 1]
```diff
@@ -17,5 +17,14 @@ module.exports = {
   currentLanguage: locale,
   isSupportedLanguage: function(lang) {
     return !!messages[lang];
+  },
+  intlDataFor: function(lang) {
```
Comment (k88hudson) [inline src/lib/i18n.js:21]: is this actually getting used?

Comment (alicoding) [inline src/lib/i18n.js:21]: Eventually we will use this, so I think it's good to have.

**Comment by thisandagain** [comment]
R+ from my POV. Thanks for the docs @alicoding !

**Pull request merged**